pyston/filters/managers.py

```diff
@@ -174,6 +175,37 @@ def filter(self, resource, qs, request):
         raise NotImplementedError
 
 
+def get_flat_lookups(q):
+    if isinstance(q, Q):
+        result = []
+        for child in q.children:
+            result += get_flat_lookups(child)
+        return result
+    else:
+        return [q[0]]
+
+
+def is_required_distinct_for_lookup(model, lookup, fail_first=False):
+    if '__' in lookup:
+        curr_lookup, next_lookup = lookup.split('__', 1)
+    else:
+        curr_lookup, next_lookup = lookup, None
+    try:
+        field = model._meta.get_field(curr_lookup)
+        if not field.is_relation:
+            return False
+        elif field.many_to_many or field.one_to_many:
+            # m2m or o2m relations can cause duplications
+            return True
+        elif not next_lookup:
+            return False
+        else:
+            return is_required_distinct_for_lookup(field.related_model, next_lookup, True)
+    except FieldDoesNotExist:
+        # For annotations is distinct automatically required
+        return fail_first
```

Comment:
if field was not found there is two posibilities:
* it is some keyword like icontains, excact (first_name__contains) - no duplications
* it is lookup for annotation, there can be duplications and annotations should be find in the first call of this methods (cannot be used name issues__anotation only annotation)